File: day3_mpi1/hello/hello.f90
program hello

use mpi
implicit none

!include "mpif.h"

integer::nprocs, ierr, myid
integer::status(mpi_status_size)

!Initialize MPI
call mpi_init(ierr)
call mpi_comm_size(mpi_comm_world, nprocs, ierr)
call mpi_comm_rank(mpi_comm_world, myid, ierr)

write(*, '("Hello World from process ",I3," /",I3)') myid, nprocs

call mpi_finalize(ierr)

stop
end program hello

File: day3_mpi1/hello/hello.c
#include "mpi.h"

int main(int argc, char* argv[])
{
  //Initialize MPI

  int nprocs, myid;
  char* name = "dear";
  if (argc == 2){
  	name = argv[1];
  }
  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nprocs);  
  MPI_Comm_rank(MPI_COMM_WORLD, &myid);
  printf("Hi %s, Hello World from process %d/%d \n", name, myid, nprocs);

  MPI_Finalize();
  
  return 0;
}
